### PR TITLE
tests: define MATCH from spread

### PR DESCRIPTION
**Pull request by zyga**
This patch teaches the test suite how to use MATCH in scripts that don't
get sourced from spread.yaml (instead of being executed as regular
programs). This makes it easier to refactor the code and use shellcheck
to look for mistakes.

Signed-off-by: Zygmunt Krynicki <zygmunt.krynicki@canonical.com>

### CODE REVIEW

File: spread.yaml [hunk 1]
```diff
@@ -347,6 +347,9 @@ prepare: |
         rmdir "$DELTA_PREFIX"
     fi
 
+    # Take the MATCH function from spread and allow our shell scripts to use it.
+    type MATCH | tail -n +2 > $TESTSLIB/spread-funcs.sh
```
Comment (chipaca) [inline spread.yaml]: isn't this redundant with including the same spread-funcs.sh below?

Comment (zyga) [reply spread.yaml]: Hmm? I must be missing something. The point of this line is to ensure we pick up the latest definition of MATCH from spread.

Comment (chipaca) [reply spread.yaml]: then why ship spread-funcs.sh at all?

Comment (chipaca) [reply spread.yaml]: maybe i should've read the comment on the file before asking that; you answer my question perfectly. Thank you.

Comment (zyga) [reply spread.yaml]: Because you can use it for local testing and spellcheck that it's all fine.

Comment (mvo5) [reply spread.yaml]: Why the tail -n +2 here ? What is in the frist lines?

File: tests/lib/spread-funcs.sh [hunk 2]
```diff
@@ -0,0 +1,24 @@
+#!/bin/bash
+
+# Define MATCH when invoked outside of spread context.
```
Comment (mvo5) [inline tests/lib/spread-funcs.sh]: Do we need to keep this function in sync with the spread implementation? If so we should probably add a comment here about it (I guess we don't have to keep it in sync its just nice if it is in sync?)

Comment (zyga) [reply tests/lib/spread-funcs.sh]: No, it is updated on each run via spread itself

File: spread.yaml (hunk 1 above)
Comment (zyga) [reply spread.yaml]: The +2 syntax means 2nd and all the following lines.

**Pull request merged**